### ISSUE TITLE
Sonarqube Plugin: Be able to have mutliple sonarqube instances

### ISSUE THREAD

**Issue by LvffY** [backstage/backstage#10674]
## Feature Suggestion

I would like to have multiple sonarqube instances plug to my backstage. 

## Possible implementation

For example, we'd like to be able to have a config like the following : 

```yaml
sonarQube:
  - baseUrl: https://your.sonarqube.instance.com
    name: myInstance ## This name could be used to refer our specific instance 
  - baseUrl: https://your.other.instance.com
    name: myOtherInstance ## This name could be used to refer our specific instance 
```

For example in my projects, I think we would need to specify wihch instance is used : 

```yaml
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: backstage
  description: |
    Backstage is an open-source developer portal that puts the developer experience first.
  annotations:
    sonarqube.org.myOtherInstance/project-key: YOUR_PROJECT_KEY
spec:
  type: library
  owner: CNCF
  lifecycle: experimental
```

## Context

In my context, we have mutliple instances of Sonarqube because we have multiple clients. Because of this context, we'd like to be able to connect our backstage app to many sonarqube instances to pull the analysis from multiple sources.

**Comment by freben** [comment]
Hi!

This sounds like it starts to warrant a dedicated backend plugin. One could take inspiration from the [jenkins backend plugin](https://github.com/backstage/backstage/tree/master/plugins/jenkins-backend) which does something similar, where users get to define the various targets and naming them, and then entities can reference those names.

**Comment by LvffY** [comment]
should have been closed by #11925, will be open again if not.

**Comment by indukuriv** [comment]
when I run the above index.ts, sonaqube.ts file after modifying based on the suggestion above. I'm getting an error TS2741: Property 'rejections' is missing in type import("../node_modules/winston/index").Logger but required in type 'import("../@backstage/plugin-sonarqube-backend/node_modules/winston/index").Logger'.       logger: env.logger

index.d.ts: rejections: RejectionHandler;
"rejections" is declared here

index.d.ts: logger: Logger;
The expected type comes from property 'logger' which is declared here on type 'RouterOptions'

Can you please suggest here
